### PR TITLE
Don't deploy the certs chart onto non-master seeds

### PR DESCRIPTION
**Pull request by nikhita**
Ref: https://github.com/kubermatic/infra/pull/141#discussion_r262002679

**Release note**:

```release-note
NONE
```

/assign @alvaroaleman

### CODE REVIEW

**Comment by nikhita** [comment]
will retest on lgtm

**Comment by xrstf** [comment]
<del>/approve</del>

**Comment by xrstf** [comment]
Don't we have Prometheus, Alertmanager and Grafana on seeds? We have never properly configured DNS records for them and don't have certs for them, but ... shouldn't we?

File: api/hack/ci/ci-deploy-dev-cloud.sh [hunk 1]
```diff
@@ -48,13 +48,22 @@ chartNamespaces=(
 function deploy {
   retry 5 kubectl apply -f ./config/kubermatic/crd/
   for chart in "${!chartNamespaces[@]}"; do
+
+    if [ "$2" == false ]; then
```
Comment (alvaroaleman) [inline api/hack/ci/ci-deploy-dev-cloud.sh]: I find this rather hard tor read. What about instead defining a `function deployMaster` and `function deploySeed`?

They can probably just wrap the existing `function deploy` but imho that will make the whole thing a lot more readable

File: api/hack/ci/ci-deploy-dev-cloud.sh [hunk 2]
```diff
@@ -54,9 +54,33 @@ function deploy {
       directory_name=$chart
     fi
 
-    retry 5 helm $1 upgrade --install --force --wait --timeout 300 \
+    retry 5 helm upgrade $1 --install --force --wait --timeout 300 \
```
Comment (alvaroaleman) [inline api/hack/ci/ci-deploy-dev-cloud.sh:57]: Shouldn't `$1` be `$chart` instead?

Comment (alvaroaleman) [reply api/hack/ci/ci-deploy-dev-cloud.sh:57]: Okay I tested it out and it seems to work but I don't understand why, do you know?

**Review by alvaroaleman — COMMENTED** [review]
`cert-manager` is superfluous for non-master seed and I don't understand _why_ but it seems to work

File: api/hack/ci/ci-deploy-dev-cloud.sh [hunk 3]
```diff
@@ -54,9 +54,33 @@ function deploy {
...
+      --set=kubermatic.rbac.image.tag=$GIT_HEAD_HASH \
+      $chart ./config/$directory_name/
+  done
+}
+
+function deploySeed {
+  retry 5 kubectl apply -f ./config/kubermatic/crd/
+  for chart in "${!chartNamespaces[@]}"; do
+      # don't deploy the certs chart onto non-master seeds
+      [ "$chart" == "certs" ] && continue
```
Comment (alvaroaleman) [inline api/hack/ci/ci-deploy-dev-cloud.sh]: `cert-manager` is also not installed into non-master seeds

File: api/hack/ci/ci-deploy-dev-cloud.sh (hunk 2 above)
Comment (nikhita) [reply api/hack/ci/ci-deploy-dev-cloud.sh:57]: > Shouldn't $1 be $chart instead?

`$1` is the extra flags that are passed to `deploySeed` and `deployMaster` like `"--kube-context=asia-east1-a-1 --tiller-namespace=kubermatic-installer"`.

The chart part is taken care of by `--namespace=${chartNamespaces[$chart]}`.

File: api/hack/ci/ci-deploy-dev-cloud.sh (hunk 3 above)
Comment (nikhita) [reply api/hack/ci/ci-deploy-dev-cloud.sh]: > cert-manager is superfluous for non-master seed and I don't understand why but it seems to work

No idea why it seems to work off the top of my mind, but taking a look.

Comment (alvaroaleman) [reply api/hack/ci/ci-deploy-dev-cloud.sh]: The "I dont understand why" was related to the "$1" not to the cert-manager chart, sorry that was a bit ambigious :S

Comment (nikhita) [reply api/hack/ci/ci-deploy-dev-cloud.sh]: fyi I've removed cert-manager but looking into why it worked.

**Pull request merged**